geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHash.java

```diff
@@ -65,50 +68,43 @@
   // have the advantage of saving us a lot of computation that would happen every time a new key was
   // added. if our internal implementation changes, these values may be incorrect. the tests will
   // catch this change. an increase in overhead should be carefully considered.
-  protected static final int BASE_REDIS_HASH_OVERHEAD = 232;
-  protected static final int HASH_MAP_VALUE_PAIR_OVERHEAD = 96;
-  protected static final int SIZE_OF_OVERHEAD_OF_FIRST_PAIR = 96;
+  protected static final int BASE_REDIS_HASH_OVERHEAD = 336;
+  protected static final int HASH_MAP_VALUE_PAIR_OVERHEAD = 48;
 
   private static final int defaultHscanSnapshotsExpireCheckFrequency =
       Integer.getInteger("redis.hscan-snapshot-cleanup-interval", 30000);
 
   private static final int defaultHscanSnapshotsMillisecondsToLive =
       Integer.getInteger("redis.hscan-snapshot-expiry", 30000);
 
-  private int HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS;
-  private int MINIMUM_MILLISECONDS_FOR_HSCAN_SNAPSHOTS_TO_LIVE;
+  private int HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS =
+      defaultHscanSnapshotsExpireCheckFrequency;
+  private int MINIMUM_MILLISECONDS_FOR_HSCAN_SNAPSHOTS_TO_LIVE =
+      defaultHscanSnapshotsMillisecondsToLive;
 
 
   @VisibleForTesting
-  public RedisHash(List<ByteArrayWrapper> fieldsToSet, int hscanSnapShotExpirationCheckFrequency,
+  public RedisHash(List<byte[]> fieldsToSet, int hscanSnapShotExpirationCheckFrequency,
       int minimumLifeForHscanSnaphot) {
-    this();
+    sizeInBytes = BASE_REDIS_HASH_OVERHEAD;
     HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS = hscanSnapShotExpirationCheckFrequency;
     MINIMUM_MILLISECONDS_FOR_HSCAN_SNAPSHOTS_TO_LIVE = minimumLifeForHscanSnaphot;
 
-    Iterator<ByteArrayWrapper> iterator = fieldsToSet.iterator();
+    hash = new Object2ObjectOpenCustomHashMap<>(fieldsToSet.size() / 2, ByteArrays.HASH_STRATEGY);
+    Iterator<byte[]> iterator = fieldsToSet.iterator();
     while (iterator.hasNext()) {
       hashPut(iterator.next(), iterator.next());
     }
   }
 
-  public RedisHash(List<ByteArrayWrapper> fieldsToSet) {
+  public RedisHash(List<byte[]> fieldsToSet) {
     this(fieldsToSet,
         defaultHscanSnapshotsExpireCheckFrequency,
         defaultHscanSnapshotsMillisecondsToLive);
   }
 
-  // for serialization
   public RedisHash() {
-    hash = new HashMap<>();
-    hScanSnapShots = new ConcurrentHashMap<>();
-    hScanSnapShotCreationTimes = new ConcurrentHashMap<>();
-
-    HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS = defaultHscanSnapshotsExpireCheckFrequency;
-
-    MINIMUM_MILLISECONDS_FOR_HSCAN_SNAPSHOTS_TO_LIVE = defaultHscanSnapshotsMillisecondsToLive;
-
-    sizeInBytes = BASE_REDIS_HASH_OVERHEAD;
+    // For serialization only!
```

Comment:
Maybe not necessary to do in this PR, but `sizeInBytes` was set in a similar way in the empty `RedisSet` constructor. We could remove it there as well.

Reply:
I moved it to the field initializer instead. This sizeInBytes is always going to get reset when this constructor is used, because it is for deserialization and we are serializing the size.